integrated_channels/lms_utils.py

```diff
@@ -14,6 +15,16 @@
 except ImportError:
     CourseGradeFactory = None
 
+try:
+    from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+except ImportError:
+    CourseOverview = None
+
+
+# Constants
+COURSE_OVERVIEW_NOT_FOUND = 'CourseOverview could not be found for course_key: {course_key}'
+COURSE_KEY_INVALID = 'Invalid course_key: {course_key}'
+
 
 def get_course_certificate(course_id, user):
     """
```

Comment:
Ah I double checked, so yes it is used like follows, either it throws that key error or returns an invalid cert

```
try:
            certificate = get_course_certificate(course_id, user)
        except InvalidKeyError:
            self._log_courseid_not_found(course_id, enterprise_enrollment, lms_user_id)
            return completed_date, grade, is_passing, percent_grade

        if not certificate:
            self._log_cert_not_found(course_id, enterprise_enrollment, lms_user_id)
            return completed_date, grade, is_passing, percent_grade
```


the impl in edx-platform of `get_certificate_for_user` looks like this so yeah looks like we are covered? 

```
try:
        cert = GeneratedCertificate.eligible_certificates.get(
            user__username=username,
            course_id=course_key
        )
    except GeneratedCertificate.DoesNotExist:
        return None

    if format_results:
        return format_certificate_for_user(username, cert)
    else:
        return cert
```

Reply:
I meant the docstring says one thing and the code does something else. Also the code behaves differently under 2 different error conditions, which means using this function will require the caller to handle None and also handle an exception. Can you make the function more consistent?

Reply:
yeah that case is interesting since it's due an importError. Perhaps it's best indicated if I actually threw an ImportError

```
if not get_certificate_for_user:
        raise ImportError("Could not load get_certificate_for_user function")
```

Reply:
Ah I see we have this pattern in other plcaes 

```
if get_certificate_for_user is None:
            raise NotConnectedToOpenEdX(
                _('To use this EnterpriseCourseEnrollmentSerializer, this package must be '
                  'installed in an Open edX environment.')
            )
```

Reply:
yeah so going with this pattern here too basically the user of this api only has to handle InvalidKeyError now, and updated docstring

Reply:
Thank you. This is much more consistent as Python has nice syntax for things like this: 
```

try:
    # do something that may fail
except IDontLikeYouException:
    # say please
except YouAreTooShortException:
    # stand on a ladder
```